hack/ci-e2e.sh

```diff
@@ -122,6 +124,24 @@ docker run --name image-server-e2e -d \
   -p 80:8080 \
   -v "${IMAGE_DIR}:/usr/share/nginx/html" nginxinc/nginx-unprivileged
 
+BMO_RELEASE_BRANCH="${BMO_RELEASE_BRANCH:-release-0.4}"
+BMO_RELEASE_ROOT="/tmp/bmo-${BMO_RELEASE_BRANCH}"
+BMO_ORIGIN_ADDRESS="https://github.com/metal3-io/baremetal-operator.git"
+export BMO_UPGRADE_FROM_IMAGE="${BMO_IMAGE_NAME}:${BMO_RELEASE_BRANCH}"
+BMO_E2E_KUSTOMIZATION="${REPO_ROOT}/config/overlays/e2e"
+
+if [[ ! -d "${BMO_RELEASE_ROOT}" ]]; then
+  git clone --branch "${BMO_RELEASE_BRANCH}" --single-branch "${BMO_ORIGIN_ADDRESS}" "${BMO_RELEASE_ROOT}"
+fi
+
+export BMO_UPGRADE_FROM_KUSTOMIZATION="${BMO_RELEASE_ROOT}/config/overlays/e2e"
+
+if [[ ! -d "${BMO_UPGRADE_FROM_KUSTOMIZATION}" ]]; then
+  mkdir -p "${BMO_UPGRADE_FROM_KUSTOMIZATION}"
+  cp -r "${BMO_E2E_KUSTOMIZATION}/"* "${BMO_UPGRADE_FROM_KUSTOMIZATION}/"
+  sed -i "s/newTag: e2e/newTag: ${BMO_RELEASE_BRANCH}/" "${BMO_UPGRADE_FROM_KUSTOMIZATION}/kustomization.yaml"
+fi
```

Comment:
Clever! We may need to change this a bit when basic-auth and TLS is added but it should be doable, don't worry about it.

Reply:
Thank you. I just realized, however, that the sed should be outside of the if case :sweat_smile: